tools/certreq/bin/certreq.sh

```diff
@@ -0,0 +1,198 @@
+#!/bin/bash
+
+# redirect stderr to stdout
+exec 2>&1
+
+# -e            Will make the whole script exit if a command fails.
+#               Append '|| true' if you have commands that can fail without being an issue.
+# -o pipefail   Causes a pipeline to produce a failure return code if any command errors.
+#set -eo pipefail
+
+# DEBUG
+if [ -n "${DEBUG+x}" ] ; then
+  set -x
+  #set -v
+fi
+
+# default values (can be changed by parameters)
+HOST="127.0.0.1"
+PORT="8000"
+USERNAME="admin"
+PASSWD="admin"
+
+readonly KEY_PAIR_FILE='private.key'
+readonly CSR_FILE='request.csr'
+readonly CA_FILE='ca.crt'
+readonly CERT_FILE='cert.pem'
+readonly TOKEN_FILE='token.jwt'
+
+readonly RED='\e[91m'
+readonly CYAN='\e[96m'
+readonly NC='\e[39m' # Default foreground color
+readonly BOLD='\e[1m'
+readonly NRML='\e[0m'
+
+function main() {
+
+  echo '============================================================='
+  echo 'This is a tool to request dojot certificates for IoT devices.'
+  echo '============================================================='
+
+  precondition
+
+  parseArgs "$@"
+
+  if [ -z "${JWT}" ]; then
+    getToken
+  fi
+
+  if [ -z "${DIVICE_IDS}" ]; then
+    getDeviceIDs
+  fi
+
+  getTrustedCA
+
+  getCertificates
+}
+
+function getToken() {
+  printf '\xE2\x8F\xB3 Obtaining access token...'
+
+  JWT=$(curl -sS -X POST "${HOST}:${PORT}/auth" \
+    -H 'Content-Type:application/json' \
+    -d "{ \"username\": \"${USERNAME}\", \"passwd\": \"${PASSWD}\" }" 2>/dev/null \
+    | gawk -v FS='"' '{print $4}')
```

Comment:
You could've used jq as well, it would be simpler, but that's OK.

Reply:
It is true! thank you very much for noticing this. That’s why code review is good, these things don’t go away :+1: